ci/scripts/r_test.sh

```diff
@@ -42,6 +42,7 @@ export _R_CHECK_TESTS_NLINES_=0
 export _R_CHECK_CRAN_INCOMING_REMOTE_=FALSE
 export _R_CHECK_LIMIT_CORES_=FALSE
 export VERSION=$(grep ^Version DESCRIPTION | sed s/Version:\ //)
+export AWS_EC2_METADATA_DISABLED=TRUE
```

Comment:
@fsaintjacques can you leave a comment explaining what this is about?

Reply:
```suggestion
# By default, aws-sdk tries to contact a non-existing local ip host
# to retrieve metadata. Disable this so that S3FileSystem tests run faster.
export AWS_EC2_METADATA_DISABLED=TRUE
```